crates/nu-cli/src/shell/filesystem_shell.rs

```diff
@@ -1212,3 +1182,53 @@ fn is_hidden_dir(dir: impl AsRef<Path>) -> bool {
         }
     }
 }
+
+#[allow(clippy::result_unwrap_used)]
+fn normalize(path: impl AsRef<Path>) -> PathBuf {
+    lazy_static::lazy_static! {
+        static ref PARENT_REGEX: Regex = Regex::new("^\\.*$").unwrap();
+    }
+
+    let mut normalized = PathBuf::new();
+    for component in path.as_ref().components() {
+        match component {
+            Component::Normal(normal) => {
+                if let Some(normal) = normal.to_str() {
+                    if PARENT_REGEX.is_match(normal) {
```

Comment:
nit: it may be a bit clearer to do

```rust
normal.chars().all(|c| c == '.')
```

It also has the benefit of not requiring another crate. Not that were opposed to bringing in a crate, but if there's a good builtin option, may as well go for it! 🙂

Reply:
You're right, I didn't know about `all`, there are a lot of iterator methods, I'll replace it. Apart from this, should I add some test cases? I'm thinking in creating a more complete canonicalize method that will replicate `realpath -m` so it doesn't necessarily fail if the path doesn't exists. I think that could fit in many operations of FilesystemShell, and possibly fix some issues.

Reply:
> Apart from this, should I add some test cases? 

The answer I will always give to that question is "yes" 🙂

Give it a shot. If you have trouble, let me know. If it's just very difficult to add a test case, we can skip it for now (we eventually want to do a pass on the code base to ensure it's better tested).

> I'm thinking in creating a more complete canonicalize method that will replicate `realpath -m` ...

Sounds like another great contribution. Feel free to do another PR for that. unless you think it's not a lot more to add to this one 🙂